applications/osb-portal/src/components/MainDrawer/index.tsx

```diff
@@ -0,0 +1,264 @@
+    const classes = useStyles();
+    const theme = useTheme();
+    const isMdUp = useMediaQuery(theme.breakpoints.up("md"));
+    const [open, setOpen] = React.useState(false);
+
+    const text= user ? `Welcome back, ${user.firstName}` :  "Lets do some science!"
+
+    const toggleDrawer = event => {
+        if (
+            event.type === "keydown" &&
+            (event.key === "Tab" || event.key === "Shift")
+        ) {
+            return;
+        }
+
+        setOpen(!open);
+    };
+
+    return (
+        <>
+        {
+        isMdUp ? null :
+            <Toolbar>
+                <IconButton
+                    color="inherit"
+                    aria-label="open drawer"
+                    edge="start"
+                    onClick={toggleDrawer}
+                    className={classes.menuButton}
+                >
+                    <MenuIcon />
+                </IconButton>
+            <Typography className={classes.helloText} variant="body2" color={secondaryColor}>
+                {text}
+            </Typography>
+         </Toolbar>
+        }
+        <Box
+            display="flex"
+            alignItems="stretch"
+            flex="1"
+            className="verticalFill"
+        >
+        <Drawer
+            variant={isMdUp ? "permanent" : "temporary"}
+            anchor="left"
+            onClose={toggleDrawer}
+            elevation={0}
+            open={open}
+            className={clsx(classes.drawer, classes.root, {
+                [classes.drawerOpen]: open,
+                [classes.drawerClose]: !open,
+            })}
+            classes={{
+                paper: clsx(classes.drawerPaper, classes.root, {
+                    [classes.drawerOpen]: open,
+                    [classes.drawerClose]: !open,
+                }),
+            }}
+        >
+            <div className={`${open ? classes.drawerContent : ""} verticalFit`}>
+                <Toolbar className={classes.toolbar}>
+                    <Typography className={classes.helloText} variant="body2" color={secondaryColor}>
+                        {text}
+                    </Typography>
+                    {
+                        !isMdUp ?
+                        <IconButton onClick={toggleDrawer}>
+                            <CloseIcon />
+                        </IconButton> : null
+                    }
+                </Toolbar>
+                <List subheader={
+                    <ListSubheader>
+                        <Typography mb={1} mt={2}>Dashboard</Typography>
+                    </ListSubheader>}>
+                    <ListItem button component='a' href='/'>
+                        <ListItemIcon>
+                            <FolderOpenIcon />
+                        </ListItemIcon>
+                        <ListItemText primary='Workspaces' />
+                    </ListItem>
+                    <ListItem button component='a' href='/repositories'>
```

Comment:
Should not be a link here, the page should not full refresh on click. Please check the useHistory hook.